Undefined refrence to ETHNET_ProgramStart(Program**) and EthnetEthernetConnection::EthnetEthernetConnection()

### Describe the bug

Building DOSBOX-X on Linux Mint (x64, version 22.3), g++ (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0, SDL Library Version 2.30.0
I get the following error after make after pull today (07/06/2026):
/usr/bin/ld: dos/libdos.a(dos_programs.o): in function `Add_VFiles(bool)':
/home/cwhite/Applications/DOSBox-X/dosbox-x/src/dos/dos_programs.cpp:10263:(.text+0x1acda): undefined reference to `ETHNET_ProgramStart(Program**)'
/usr/bin/ld: misc/libmisc.a(ethernet.o): in function `OpenEthernetConnection(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
/home/cwhite/Applications/DOSBox-X/dosbox-x/src/misc/ethernet.cpp:68:(.text+0x516): undefined reference to `EthnetEthernetConnection::EthnetEthernetConnection()'
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:497: dosbox-x] Error 1
make[3]: Leaving directory '/home/cwhite/Applications/DOSBox-X/dosbox-x/src'
make[2]: *** [Makefile:557: all-recursive] Error 1
make[2]: Leaving directory '/home/cwhite/Applications/DOSBox-X/dosbox-x/src'
make[1]: *** [Makefile:411: all-recursive] Error 1
make[1]: Leaving directory '/home/cwhite/Applications/DOSBox-X/dosbox-x'
make: *** [Makefile:344: all] Error 2


### Steps to reproduce the behaviour

make clean
./autogen.sh
./configure --enable-sdl2
make

### Expected behavior

_No response_

### What operating system(s) this bug have occurred on?

Linux Mint (x64, version 22.3)

### What version(s) of DOSBox-X have this bug?

commit bbd900626dcde3e639570a6ea42e351c6d4095cf

### Used configuration

```ini

```

### Output log

```text
/usr/bin/ld: dos/libdos.a(dos_programs.o): in function `Add_VFiles(bool)':
/home/cwhite/Applications/DOSBox-X/dosbox-x/src/dos/dos_programs.cpp:10263:(.text+0x1acda): undefined reference to `ETHNET_ProgramStart(Program**)'
/usr/bin/ld: misc/libmisc.a(ethernet.o): in function `OpenEthernetConnection(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
/home/cwhite/Applications/DOSBox-X/dosbox-x/src/misc/ethernet.cpp:68:(.text+0x516): undefined reference to `EthnetEthernetConnection::EthnetEthernetConnection()'
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:497: dosbox-x] Error 1
make[3]: Leaving directory '/home/cwhite/Applications/DOSBox-X/dosbox-x/src'
make[2]: *** [Makefile:557: all-recursive] Error 1
make[2]: Leaving directory '/home/cwhite/Applications/DOSBox-X/dosbox-x/src'
make[1]: *** [Makefile:411: all-recursive] Error 1
make[1]: Leaving directory '/home/cwhite/Applications/DOSBox-X/dosbox-x'
make: *** [Makefile:344: all] Error 2
```

### Additional information

_No response_

### Have you checked that no similar bug report(s) exist?

- [x] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [x] I agree to follow the code of conduct and the contributing guidelines.

--- a/src/dos/dos_programs.cpp
+++ b/src/dos/dos_programs.cpp
@@ -10262,8 +10262,10 @@ void Add_VFiles(bool usecp) {
 # endif
 	if (addne2k) {
 		VFILE_RegisterBuiltinFileBlob(bfb_NE2000_COM, "/SYSTEM/");
-		PROGRAMS_MakeFile("ETHNET.COM",ETHNET_ProgramStart,"/SYSTEM/");
-	}
+#if defined(C_SDL_NET) || defined(C_SDL2_NET)
+        PROGRAMS_MakeFile("ETHNET.COM",ETHNET_ProgramStart,"/SYSTEM/");
+#endif
+    }
 	if (addovl) VFILE_RegisterBuiltinFileBlob(bfb_GLIDE2X_OVL, "/SYSTEM/");
 #endif
 

--- a/src/dos/dos_programs.cpp
+++ b/src/dos/dos_programs.cpp
@@ -8083,7 +8083,9 @@ void SERIAL_ProgramStart(Program * * make);
 void CONFIG_ProgramStart(Program * * make);
 #endif
 void IPXNET_ProgramStart(Program * * make);
+#if (defined(C_SDL_NET) || defined(C_SDL2_NET))
 void ETHNET_ProgramStart(Program * * make);
+#endif
 void A20GATE_ProgramStart(Program * * make);
 void CGASNOW_ProgramStart(Program * * make);
 void PARALLEL_ProgramStart(Program * * make);

--- a/src/misc/ethernet.cpp
+++ b/src/misc/ethernet.cpp
@@ -38,8 +38,10 @@ EthernetConnection* OpenEthernetConnection(std::string backendstr)
         backend = "slirp";
 #elif defined(C_PCAP)
         backend = "pcap";
-#else
+#elif defined(C_SDL_NET) || defined(C_SDL2_NET)
         backend = "ethnet";
+#else
+        backend = "none";
 #endif
     } else {
         backend = backendstr;

--- a/src/misc/ethernet.cpp
+++ b/src/misc/ethernet.cpp
@@ -62,6 +62,7 @@ EthernetConnection* OpenEthernetConnection(std::string backendstr)
         if (!conn->Initialize(settings)) { delete conn; conn = NULL; }
     }
 #endif
+#if defined(C_SDL_NET) || defined(C_SDL2_NET)
     if (backendstr == "auto" && !conn) backend = "ethnet";
     if (backend == "ethnet") {
         assert(conn == NULL);
@@ -69,7 +70,7 @@ EthernetConnection* OpenEthernetConnection(std::string backendstr)
         settings = control->GetSection("ethernet, pcap");//NTS: The ethnet uses no settings, but there is an assert below to ensure settings != NULL
         if (!conn->Initialize(settings)) { delete conn; conn = NULL; }
     }
-
+#endif
     if (backendstr == "auto" && !conn) backend = "nothing";
     if (backend == "nothing") {
         assert(conn == NULL);

--- a/src/misc/ethernet_ethnet.cpp
+++ b/src/misc/ethernet_ethnet.cpp
@@ -19,6 +19,12 @@
 #include "config.h"
 
 #if defined(C_SDL_NET) || defined(C_SDL2_NET)
+#ifdef _WIN32
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#else
+#include <arpa/inet.h>
+#endif
 #include "ethernet_ethnet.h"
 #include "dosbox.h"
 #include "logging.h"

